### PR TITLE
Fixed crashes when NatDevice.GetExternalIP() fails

### PR DESCRIPTION
**Pull request by Mailaender**
Rule of thumb when working with Mono.NAT: try/catch everything.

Fixes https://github.com/OpenRA/OpenRA/issues/9405.
Fixes https://github.com/OpenRA/OpenRA/issues/9671.

Compare https://github.com/OpenRA/OpenRA/pull/2993.

It isn't obvious, but somehow even simply getting an external IP address can cause MappingExceptions?!

### CODE REVIEW

File: OpenRA.Game/Network/UPnP.cs [hunk 1]
```diff
@@ -116,5 +116,22 @@ public static void RemovePortforward()
...
+
+		public static IPAddress GetExternalIP()
+		{
+			if (natDevice == null)
+				return null;
+
+			try
+			{
+				var ip = natDevice.GetExternalIP();
+				return ip;
```
Comment (RoosterDragon) [inline OpenRA.Game/Network/UPnP.cs]: Maybe `return natDevice.GetExternalIP();`?

**Comment by RoosterDragon** [comment]
:+1: Seems sensible.

File: OpenRA.Game/Network/UPnP.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Game/Network/UPnP.cs]: Indeed.

**Comment by abcdefg30** [comment]
Looks good to me, too. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/33a9ee760ffb4035fae438470e4c144443a5efe5...4bd64db81c04649e74fbaf78415937c10fed346b)

**Pull request merged**